Libraries/Opc.Ua.Server/Diagnostics/MonitoredNode.cs

```diff
@@ -281,39 +228,31 @@ public void OnReportEvent(ISystemContext context, NodeState node, IFilterTarget
         /// <param name="changes">The mask indicating what changes have occurred.</param>
         public void OnMonitoredNodeChanged(ISystemContext context, NodeState node, NodeStateChangeMasks changes)
         {
-            lock (NodeManager.Lock)
```

Comment:
We need to enshure Data Changes reach each monitored Item in the right order i would sugesst a semaphore only for this method. No need to lock the entire node manager to enshure this behaviour (the node changes reach this method anyway)